### PR TITLE
feat: Add Prow to service catalog

### PR DESCRIPTION
**Pull request by tumido**
SSIA

/cc @SamoKopecky

### CODE REVIEW

File: service-catalog/components/prow-hook.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: prow-hook
+  title: Hook
+  description: Receives all the web hooks from GitHub. It is a stateless server that listens for GitHub webhooks and dispatches them to the appropriate plugins.
```
Comment (HumairAK) [inline service-catalog/components/prow-hook.yaml:6]:
```suggestion
  description: Receives all the webhooks from GitHub. It is a stateless server that listens for GitHub webhooks and dispatches them to the appropriate plugins.
```

Comment (tumido) [reply service-catalog/components/prow-hook.yaml:6]: this is all taken from upstream.. but :shrug: why not...

**Pull request merged**